lib/src/util/dart_type_utilities.dart

```diff
@@ -595,6 +558,48 @@ class InterfaceTypeDefinition {
   }
 }
 
+extension ElementExtensions on Element {
```

Comment:
Consider moving these into `ast.dart`?  Or maybe introducing a new `extensions.dart` like we have in analyzer?

Reply:
Ah ha I hadn't seen that file. Done!